com.reprezen.swagedit/src/com/reprezen/swagedit/json/references/JsonReference.java

```diff
@@ -36,16 +36,19 @@
     private final boolean absolute;
     private final boolean local;
     private final Object source;
+    private final boolean containsWarning;
 
     private JsonNode resolved;
     private JsonDocumentManager manager = JsonDocumentManager.getInstance();
 
-    JsonReference(URI uri, JsonPointer pointer, boolean absolute, boolean local, Object source) {
+    JsonReference(URI uri, JsonPointer pointer, boolean absolute, boolean local, boolean containsWarning,
+            Object source) {
```

Comment:
We already reached 6 constructor parameter. In three usages of four we use the same values for the first five params: `new JsonReference(null, null, false, false, false, node)`.
If we ever decide to add another constructor parameter it would be good to see how we can make it more readable, e.g. use fluent builder API or create new constructors for most commonly used cases.